WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,139 @@
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <div id="scroller-anchor"></div>
+  <div id="comp-query" class="col-md-12">
+    <% if params[:display] == "List" %>
+      <ul>
+        <% if @in_progress_competitions.length > 0 %>
+          <li><a href="#current">Competitions in progress</a></li>
+        <% else %>
+          <li>There are no competitions today.</li>
+        <% end %>
+
+        <% if @upcoming_competitions.length > 0 %>
+          <li><a href="#upcoming">Upcoming competitions</a></li>
+        <% else %>
+          <li>There are no upcoming competitions selected.</li>
+        <% end %>
+
+        <% if @past_competitions.length > 0 %>
+          <li><a href="#past">Past competitions</a></li>
+        <% else %>
+          <li>There are no past competitions selected.</li>
+        <% end %>
+      </ul>
+    <% end %>
+
+    <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+      <%= render 'index_form' %>
+    <% end %>
+  </div>
+    <% if @competitions.length == 0 %>
+      <div class="alert alert-warning">No competitions found</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <div class="row competition-list">
+          <div class="col-md-12" id="current">
+            <% if @in_progress_competitions.length > 0 %>
+              <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+              <% end %>
+          </div>
+
+          <div class="col-md-6" id="upcoming">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+          <div class="col-md-6" id="past">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions" %>
+          </div>
+        </div>
+      <% else %>
+        <div class="row competition-list">
+          <div class="col-xs-12 col-md-12">
+            <div id="comps-map"></div>
+          </div>
+        </div>
+      <% end %>
     <% end %>
-  </table>
 </div>
+
+<script>
+function moveScroller() {
+  var move = function() {
+    var st = $(window).scrollTop();
+    var ot = $("#scroller-anchor").offset().top;
+    var s = $("#comp-query");
+    if(st > ot) {
+        s.css({
+            position: "fixed",
+            top: "0px",
+            "z-index": 10,
+            "background-color": "white"
+        });
+    } else {
+        if(st <= ot) {
+            s.css({
+                position: "relative",
+                top: ""
+            });
+        }
+    }
+  };
+  $(window).scroll(move);
+  move();
+}
+
+function mapSize() {
```

Comment:
Can you add a comment to this method that points to https://github.com/cubing/worldcubeassociation.org/issues/356?